### DISCUSSION TITLE
Support of AWS' DocumentDB ?

### DISCUSSION THREAD

**Discussion by TechnicalMercenary** [micronaut-projects/micronaut-core#7618] [Q&A]
AWS has a 'mongo like' product called DocumentDB.  It is close enough to mongo that standard mongo drivers can integrate with it.

When a DocumentDb database is created, AWS provides a TLS Certificate to secure communication.

Searching the web and this discussion forum, I have not seen any details regarding documentDB or certificate support in general.

Is using a TLS cert to secure a db connection possible ?

**Comment by dstepanov** [comment]
Maybe there is something in the Micronaut AWS module documentation https://micronaut-projects.github.io/micronaut-aws/latest/guide/

**Comment by graemerocher** [comment]
It is planned to potentially support DocumentDB one day as part of Micronaut Data MongoDB, perhaps later this year.

**Reply by o-shevchenko** [reply]
Is it possible to configure a certificate for Mongo Data?

**Reply by o-shevchenko** [reply]
Should we use something like: 
```
 System.setProperty("javax.net.ssl.trustStore", "/home/username/mongostore");
  System.setProperty("javax.net.ssl.trustStorePassword", "ilikeamiga");
  ```
  to set a certificate for Mongo Driver to connect to DocumentDB?
  https://docs.aws.amazon.com/documentdb/latest/developerguide/connect_programmatically.html

**Reply by dstepanov** [reply]
Did it work for you? You might need to enable SSL for Mongo driver

**Reply by o-shevchenko** [reply]
>Did it work for you?

Haven't tried yet. 
>You might need to enable SSL for Mongo driver

I don't see any tls or certificate mentioned in Mongo Client code. What configuration do you mean?

**Reply by dstepanov** [reply]
If you look inside `DefaultMongoConfiguration` you can see there is a builder for SSL configuration, so it should be possible to write:
```yaml
mongodb:
   uri: xxx
   ssl: true
```

**Reply by o-shevchenko** [reply]
then yes, I set it.
But the problem is that there isn't an option to set the certificate path for MongoDB.
Something like in pymongo: https://pymongo.readthedocs.io/en/stable/examples/tls.html
```python
client = pymongo.MongoClient('example.com',
                             tls=True,
                             tlsCertificateKeyFile='/path/to/client.pem')
```

**Reply by o-shevchenko** [reply]
I successfully connected to DocumentDB using the following configurations:
```
mongodb:
  uri: mongodb://<USER>:<PASSWORD>@<HOST>:<PORT>/<DBNAME>?ssl=true&replicaSet=rs0&readPreference=secondaryPreferred&retryWrites=false
```
SSL cert should be added to Java truststore (https://docs.aws.amazon.com/documentdb/latest/developerguide/connect_programmatically.html).
I added two certificates:
``` 
keytool -importcert -noprompt -file /tmp/rds-ca-2019-root.pem -keystore /usr/local/java/lib/security/cacerts -storepass changeit -alias rds-ca-2019-root
keytool -importcert -noprompt -file /tmp/rds-combined-ca-bundle.pem -keystore /usr/local/java/lib/security/cacerts -storepass changeit -alias rds-combined-ca-bundle
```
